service/frontend/workflow_handler.go

```diff
@@ -3403,64 +3403,74 @@ func (wh *WorkflowHandler) UpdateWorkflowExecution(
 ) (_ *workflowservice.UpdateWorkflowExecutionResponse, retError error) {
 	defer log.CapturePanic(wh.logger, &retError)
 
+	if err := wh.prepareUpdateWorkflowRequest(request); err != nil {
+		return nil, err
+	}
+
+	nsID, err := wh.namespaceRegistry.GetNamespaceID(namespace.Name(request.GetNamespace()))
+	if err != nil {
+		return nil, err
+	}
+
+	histResp, err := wh.historyClient.UpdateWorkflowExecution(ctx, &historyservice.UpdateWorkflowExecutionRequest{
+		NamespaceId: nsID.String(),
+		Request:     request,
+	})
+
+	return histResp.GetResponse(), err
+}
+
+func (wh *WorkflowHandler) prepareUpdateWorkflowRequest(
+	request *workflowservice.UpdateWorkflowExecutionRequest,
+) error {
 	if request == nil {
-		return nil, errRequestNotSet
+		return errRequestNotSet
 	}
 
 	if err := validateExecution(request.GetWorkflowExecution()); err != nil {
-		return nil, err
+		return err
 	}
 
 	if request.GetRequest().GetMeta() == nil {
-		return nil, errUpdateMetaNotSet
+		return errUpdateMetaNotSet
 	}
 
 	if len(request.GetRequest().GetMeta().GetUpdateId()) > wh.config.MaxIDLengthLimit() {
-		return nil, errUpdateIDTooLong
+		return errUpdateIDTooLong
 	}
 
 	if request.GetRequest().GetMeta().GetUpdateId() == "" {
 		request.GetRequest().GetMeta().UpdateId = uuid.New()
 	}
 
 	if request.GetRequest().GetInput() == nil {
-		return nil, errUpdateInputNotSet
+		return errUpdateInputNotSet
 	}
 
 	if request.GetRequest().GetInput().GetName() == "" {
-		return nil, errUpdateNameNotSet
+		return errUpdateNameNotSet
 	}
 
 	if request.GetWaitPolicy() == nil {
 		request.WaitPolicy = &updatepb.WaitPolicy{}
 	}
 
-	nsID, err := wh.namespaceRegistry.GetNamespaceID(namespace.Name(request.GetNamespace()))
-	if err != nil {
-		return nil, err
-	}
-
```

Comment:
This is the only behavior change. I believe it's safe to remove this from here as I've verified that `EnableUpdateWorkflowExecution` can handle a non-existing namespace.

Reply:
Is there a reason we don't want this check in other users of `prepareUpdateWorkflowRequest`?

Reply:
I suppose we'll need to grab the nsid in any other consumer regardless

Reply:
It's my believe that this should be down higher up in the stack (as it happens here and will for the next use case which is MultiOperation API), ie by the caller. The check is not particular to the Update request, it needs to happen on any kind of request.

Reply:
ie it's more for our internal plumbing, not request-specific.